connectors/sources/sharepoint_online.py

```diff
@@ -351,12 +359,34 @@ async def scroll_delta_url(self, url):
 
     async def _get_json(self, absolute_url):
         self._logger.debug(f"Fetching url: {absolute_url}")
-        async with self._call_api(absolute_url) as resp:
+        async with self._get(absolute_url) as resp:
             return await resp.json()
 
     @asynccontextmanager
     @retryable_aiohttp_call(retries=DEFAULT_RETRY_COUNT)
-    async def _call_api(self, absolute_url):
+    async def _post(self, absolute_url, payload=None):
+        try:
+            token = await self._api_token.get()
+            headers = {"authorization": f"Bearer {token}"}
+            self._logger.debug(f"Calling Sharepoint Endpoint: {absolute_url}")
+
+            async with self._http_session.post(
+                absolute_url, headers=headers, json=payload
+            ) as resp:
+                yield resp
+
+            return
```

Comment:
If it only yield one item, do we need to use `yield`?

Reply:
AFAIU yes as it's an `@asynccontextmanager`.